ex2_system/source/diagnostic/diagnosic.c

```diff
@@ -20,38 +20,113 @@
 
 #include <FreeRTOS.h>
 #include <os_task.h>
+#include <os_semphr.h>
+#include "uhf.h"
+#include "eps.h"
+#include "system.h"
 
-static void *diagnostic_daemon(void *pvParameters);
+static void uhf_watchdog_daemon(void *pvParameters);
 SAT_returnState start_diagnostic_daemon(void);
 
+const unsigned int mutex_timeout = pdMS_TO_TICKS(100);
+static TickType_t prv_watchdog_delay = 1 * ONE_MINUTE; // 3 minutes

```

Comment:
@nsorense4 Please merge master and resolve conflicts. Change or remove this comment if you want to use 1 minute instead of 3.

Reply:
Done